### ISSUE TITLE
0.77 dev accuracy on quora question pairs, but 0.91 is the state of the art

### ISSUE THREAD

**Issue by weizier** [IndicoDataSolutions/finetune#72]
I ran quora_similarity.py, and didn't modify any codes, then I got 0.77 accuracy for 0.4 class balance, any ideas for this? Thanks.

**Comment by benleetownsend** [comment]
The demo script runs with only the first 5k (out of 400k) rows of data.  I’m currently running the full dataset so i’ll update you with the results once it finishes.

**Comment by benleetownsend** [comment]
Just a quick update, got Out of Memory Error before it even started (batch size too big :cry: ) . I'll rerun overnight and should have some results for you tomorrow. If you beat me too it, id be interested to hear what you find.

**Comment by weizier** [comment]
How is it going on?

**Comment by madisonmay** [comment]
@weizier going to give it another run overnight and see where things up.  Will keep you posted.

**Comment by benleetownsend** [comment]
@weizier Sorry for going quiet for a few days. I was able to run this over the weekend, unfortunately, the results were worse than at 5k rows. Spent a few hours today trying to understand this but haven't come up with anything particularly concrete. I will run again tonight with a few little bug fixes and see what I get. Hopefully I will have more info for you tomorrow.

**Comment by benleetownsend** [comment]
We recently were able to figure out some problems that were causing issues with the Quora dataset, apologies for the long delay. 

```
dataset = QuoraDuplicate().dataframe
    model = Comparison(low_memory_mode=True, n_epochs=5, batch_size=32, early_stopping_steps=10000)
```
Replacing these two lines from finetune/datasets/quora_similarity.py to fit on the whole dataset, with a larger batch size, yields 90% accuracy.

Any discrepancies between this and other results is due to the train and test sets being split from the train set meaning 30% of the data is held out for test causing a smaller train set and a different test set. 

Hopefully this answers your query. I will close this issue for now as it appears to be solved, if you have any further questions feel free to reopen.
